src/IterableCodeExtractor.php

```diff
@@ -176,7 +176,7 @@ protected static function containsMatchingChildren( SplFileInfo $dir, array $mat
 			// Or the start of the matcher until the first wildcard matches the start of the path.
 			if (
 				( '' !== $root_relative_path && 0 === strpos( $base, $root_relative_path ) ) ||
-				0 === strpos( $root_relative_path, $base )
+				( '' !== $base && 0 === strpos( $root_relative_path, $base ) )
```

Comment:
@swissspidy This now saves against the thrown notice, but I'm wondering whether it doesn't now exclude valid matches.

If `$base` is empty, should a match be possible here?

Reply:
The old code, without my change, in context, was:

```php
	if (
		( '' !== $root_relative_path && 0 === strpos( $base, $root_relative_path ) ) ||
		0 === strpos( $root_relative_path, $base )
	) {
		return true;
	}
}
return false;
```

Written another way: 

```php
	if (
		$CONDITION_1 ||
		$CONDITION_2
	) {
		return true;
	}
}
return false;
```

My change only suppresses the warning:

```php
$CONDITION_2 = strpos( 'node_modules', '' );
var_dump($CONDITION_2);
// false
```

```php
$CONDITION_2 = '' !== '' ;
var_dump($CONDITION_2);
// false
```

> whether it doesn't now exclude valid matches. If $base is empty, should a match be possible here?

My change doesn't change how the code works, it only suppresses the PHP Warning.

If a match is possible, and the testing suite doesn't catch it, then that's beyond the scope of my PR IMHO.

Reply:
Yes, it was clear how the logic works. However, the PR should ideally be about eliminating the root cause of a bug, not just obfuscating it.

In this case, the root cause is that one possible scenario was apparently not being considered by the current logic. The notice as such is valuable as it shows us we apparently forgot something. If we now just hide the notice, the possible bug might still be in there and forgotten because the symptom is being hidden.

This is why I'd like some feedback by @swissspidy, who built the original code, as I'm unsure how what an empty base in this case would signify.

Reply:
Hmm I think my initial work was pretty much rewritten in #104. So maybe @herregroen knows more?

The warning implies that `$path_or_file` is an empty string and I wonder why this happens. If it's because of some edge case thing I don't think just `break`ing early would cause any troubles.

Reply:
Maybe this helps? I can reproduce every time by running:

```
wp i18n make-pot . languages/plugin.pot --include="*.blade.php" --exclude="vendor,symbionts" --headers='{"Report-Msgid-Bugs-To":"https://github.com/namespace/plugin/"}
```

Reply:
Ah, indeed! My bad.